controller_manager/src/controller_manager.cpp

```diff
@@ -53,61 +58,787 @@ ControllerManager::load_controller(
+      "strictness level of controller_manager_msgs::SwitchController::STRICT "
+      "(%d) or ::BEST_EFFORT (%d). Defaulting to ::BEST_EFFORT.",
+      controller_manager_msgs::srv::SwitchController::Request::STRICT,
+      controller_manager_msgs::srv::SwitchController::Request::BEST_EFFORT);
+    strictness = controller_manager_msgs::srv::SwitchController::Request::BEST_EFFORT;
+  }
+
+  RCLCPP_DEBUG(get_logger(), "switching controllers:");
+  for (const auto & controller : start_controllers) {
+    RCLCPP_DEBUG(get_logger(), "- starting controller '%s'", controller.c_str());
+  }
+  for (const auto & controller : stop_controllers) {
+    RCLCPP_DEBUG(get_logger(), "- stopping controller '%s'", controller.c_str());
+  }
+
+  // list all controllers to stop
+  for (const auto & controller : stop_controllers) {
+    controller_interface::ControllerInterfaceSharedPtr ct = get_controller_by_name(controller);
+    if (!ct.get()) {
+      if (strictness == controller_manager_msgs::srv::SwitchController::Request::STRICT) {
+        RCLCPP_ERROR(
+          get_logger(),
+          "Could not stop controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+        stop_request_.clear();
+        return controller_interface::return_type::ERROR;
+      } else {
+        RCLCPP_DEBUG(
+          get_logger(),
+          "Could not stop controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+      }
+    } else {
+      RCLCPP_DEBUG(
+        get_logger(),
+        "Found controller '%s' that needs to be stopped in list of controllers",
+        controller.c_str());
+      stop_request_.push_back(ct);
+    }
+  }
+  RCLCPP_DEBUG(get_logger(), "Stop request vector has size %i", (int)stop_request_.size());
+
+  // list all controllers to start
+  for (const auto & controller : start_controllers) {
+    controller_interface::ControllerInterfaceSharedPtr ct = get_controller_by_name(controller);
+    if (!ct.get()) {
+      if (strictness == controller_manager_msgs::srv::SwitchController::Request::STRICT) {
+        RCLCPP_ERROR(
+          get_logger(),
+          "Could not start controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+        stop_request_.clear();
+        start_request_.clear();
+        return controller_interface::return_type::ERROR;
+      } else {
+        RCLCPP_DEBUG(
+          get_logger(),
+          "Could not start controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+      }
+    } else {
+      RCLCPP_DEBUG(
+        get_logger(), "Found controller '%s' that needs to be started in list of controllers",
+        controller.c_str());
+      start_request_.push_back(ct);
+    }
+  }
+  RCLCPP_DEBUG(get_logger(), "Start request vector has size %i", (int)start_request_.size());
```

Comment:
This should be a free-function or a lambda, it's the same code as for the search for controllers to stop and thus should be de-duplicated.

Reply:
Done, `list_controllers` lambda in `switch_controller()`